### ISSUE TITLE
julia 0.6.2 fails to build on Ubuntu 16.04

### ISSUE THREAD

**Issue by adamryczkowski** [spack/spack#7857]
### Steps to reproduce the issue

1. Update spack to version 

```commit c8e3c94c6ffa8a5f24423e53958ee71cf98a60d6
Author: 健美猫 <weijianwen@gmail.com>
Date:   Fri Apr 20 01:43:12 2018 +0800

    self has no attribute named satisfies. (#7826)
```

2. Then in the console:

```
spack --verbose --debug install -j 1 julia
==> Reading config file /home/Adama-docs/Adam/linux/tmp/spack/etc/spack/defaults/packages.yaml
==> READ LOCK: /home/adam/.spack/cache/providers/.builtin-index.yaml.lock[0:0] [Acquiring]
==> READ LOCK: /home/adam/.spack/cache/providers/.builtin-index.yaml.lock[0:0] [Acquired]
==> READ LOCK: /home/adam/.spack/cache/providers/.builtin-index.yaml.lock[0:0] [Released]
==> Reading config file /home/adam/.spack/linux/compilers.yaml
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[2624999854097962482:1] [Acquiring]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[2624999854097962482:1] [Acquired]
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/lock[0:0] [Acquiring]
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/lock[0:0] [Acquired]
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/lock[0:0] [Released]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[2624999854097962482:1] [Released]
==> link /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/julia-0.6.2-jdn352z55qv6kjotkpxczr2phklrtxu7 -> /tmp/adam/spack-stage/spack-stage-1q4cgpcb
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[2624999854097962482:1] [Acquiring]
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[2624999854097962482:1] [Acquired]
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[2624999854097962482:1] [Released]
==> Installing julia dependencies
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[7601515057099834515:1] [Acquiring]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[7601515057099834515:1] [Acquired]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[7601515057099834515:1] [Released]
==> link /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/libsigsegv-2.11-2l6agvx2oaesnpa5m6ete54fbzv7loem -> /tmp/adam/spack-stage/spack-stage-ad2q_8c0
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[7601515057099834515:1] [Acquiring]
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[7601515057099834515:1] [Acquired]
==> libsigsegv is already installed in /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-6.3.0/libsigsegv-2.11-2l6agvx2oaesnpa5m6ete54fbzv7loem
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[7601515057099834515:1] [Released]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[6597854822771567266:1] [Acquiring]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[6597854822771567266:1] [Acquired]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[6597854822771567266:1] [Released]
==> link /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/m4-1.4.18-w4qjiar5znkuk7huxveili63ymzckbvt -> /tmp/adam/spack-stage/spack-stage-a67359q_
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[6597854822771567266:1] [Acquiring]
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[6597854822771567266:1] [Acquired]
==> m4 is already installed in /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-6.3.0/m4-1.4.18-w4qjiar5znkuk7huxveili63ymzckbvt
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[6597854822771567266:1] [Released]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[30598824919622713:1] [Acquiring]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[30598824919622713:1] [Acquired]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[30598824919622713:1] [Released]
==> link /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/bison-3.0.4-admwv4btyhehgf5cttnzdul35sklzcar -> /tmp/adam/spack-stage/spack-stage-fswsr221
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[30598824919622713:1] [Acquiring]
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[30598824919622713:1] [Acquired]
==> bison is already installed in /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-6.3.0/bison-3.0.4-admwv4btyhehgf5cttnzdul35sklzcar
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[30598824919622713:1] [Released]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[408819729993433793:1] [Acquiring]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[408819729993433793:1] [Acquired]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[408819729993433793:1] [Released]
==> link /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/bzip2-1.0.6-bnmnnqn63qcyh22d5ucs2c42feet7thp -> /tmp/adam/spack-stage/spack-stage-jhur8wy8
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[408819729993433793:1] [Acquiring]
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[408819729993433793:1] [Acquired]
==> bzip2 is already installed in /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-6.3.0/bzip2-1.0.6-bnmnnqn63qcyh22d5ucs2c42feet7thp
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[408819729993433793:1] [Released]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[337469381091415525:1] [Acquiring]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[337469381091415525:1] [Acquired]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[337469381091415525:1] [Released]
==> link /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/pkgconf-1.4.0-bfo52otnbsv4uptg3vufgvkpzsl7resc -> /tmp/adam/spack-stage/spack-stage-3e3rz9kb
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[337469381091415525:1] [Acquiring]
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[337469381091415525:1] [Acquired]
==> pkgconf is already installed in /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-6.3.0/pkgconf-1.4.0-bfo52otnbsv4uptg3vufgvkpzsl7resc
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[337469381091415525:1] [Released]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[3890735237286920222:1] [Acquiring]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[3890735237286920222:1] [Acquired]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[3890735237286920222:1] [Released]
==> link /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/xz-5.2.3-np6vmkzlfiad2er2ych3alsf3vihutbl -> /tmp/adam/spack-stage/spack-stage-0mtu03ye
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[3890735237286920222:1] [Acquiring]
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[3890735237286920222:1] [Acquired]
==> xz is already installed in /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-6.3.0/xz-5.2.3-np6vmkzlfiad2er2ych3alsf3vihutbl
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[3890735237286920222:1] [Released]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[8010208185367732976:1] [Acquiring]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[8010208185367732976:1] [Acquired]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[8010208185367732976:1] [Released]
==> link /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/zlib-1.2.11-3zj7htztrjc6cnowfdwcrw3univs3bnz -> /tmp/adam/spack-stage/spack-stage-7s08cmyp
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[8010208185367732976:1] [Acquiring]
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[8010208185367732976:1] [Acquired]
==> zlib is already installed in /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-6.3.0/zlib-1.2.11-3zj7htztrjc6cnowfdwcrw3univs3bnz
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[8010208185367732976:1] [Released]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[3264708561876847106:1] [Acquiring]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[3264708561876847106:1] [Acquired]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[3264708561876847106:1] [Released]
==> link /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/libxml2-2.9.4-lkosmr43wekalxeobwfdx7bam37ex4fa -> /tmp/adam/spack-stage/spack-stage-2wky99_q
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[3264708561876847106:1] [Acquiring]
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[3264708561876847106:1] [Acquired]
==> libxml2 is already installed in /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-6.3.0/libxml2-2.9.4-lkosmr43wekalxeobwfdx7bam37ex4fa
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[3264708561876847106:1] [Released]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[6588140763506358323:1] [Acquiring]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[6588140763506358323:1] [Acquired]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[6588140763506358323:1] [Released]
==> link /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/ncurses-6.0-w3ny4phgejygng2zgd3tpo443olpxlr2 -> /tmp/adam/spack-stage/spack-stage-iobr_s_8
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[6588140763506358323:1] [Acquiring]
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[6588140763506358323:1] [Acquired]
==> ncurses is already installed in /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-6.3.0/ncurses-6.0-w3ny4phgejygng2zgd3tpo443olpxlr2
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[6588140763506358323:1] [Released]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[4339079054668997740:1] [Acquiring]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[4339079054668997740:1] [Acquired]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[4339079054668997740:1] [Released]
==> link /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/tar-1.29-pbxqfustyrenqmpm3xodztkuijvhz75w -> /tmp/adam/spack-stage/spack-stage-r3e5p2y2
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[4339079054668997740:1] [Acquiring]
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[4339079054668997740:1] [Acquired]
==> tar is already installed in /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-6.3.0/tar-1.29-pbxqfustyrenqmpm3xodztkuijvhz75w
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[4339079054668997740:1] [Released]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[4245136208677497926:1] [Acquiring]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[4245136208677497926:1] [Acquired]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[4245136208677497926:1] [Released]
==> link /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/gettext-0.19.8.1-oxjydr52xxmiy4veh4fwn4tymu5ux7iy -> /tmp/adam/spack-stage/spack-stage-vvdcn3dg
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[4245136208677497926:1] [Acquiring]
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[4245136208677497926:1] [Acquired]
==> gettext is already installed in /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-6.3.0/gettext-0.19.8.1-oxjydr52xxmiy4veh4fwn4tymu5ux7iy
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[4245136208677497926:1] [Released]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[5348130764037331580:1] [Acquiring]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[5348130764037331580:1] [Acquired]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[5348130764037331580:1] [Released]
==> link /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/help2man-1.47.4-srymcmcfse2pthl3yspptryvbuomd32m -> /tmp/adam/spack-stage/spack-stage-vpj8pnmw
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[5348130764037331580:1] [Acquiring]
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[5348130764037331580:1] [Acquired]
==> help2man is already installed in /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-6.3.0/help2man-1.47.4-srymcmcfse2pthl3yspptryvbuomd32m
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[5348130764037331580:1] [Released]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[5999189374454010171:1] [Acquiring]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[5999189374454010171:1] [Acquired]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[5999189374454010171:1] [Released]
==> link /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/flex-2.6.4-u2bm3zxjiirhmcfcbbfllaj3m56235kk -> /tmp/adam/spack-stage/spack-stage-aof9qq3d
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[5999189374454010171:1] [Acquiring]
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[5999189374454010171:1] [Acquired]
==> flex is already installed in /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-6.3.0/flex-2.6.4-u2bm3zxjiirhmcfcbbfllaj3m56235kk
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[5999189374454010171:1] [Released]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[1860118916110345305:1] [Acquiring]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[1860118916110345305:1] [Acquired]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[1860118916110345305:1] [Released]
==> link /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/binutils-2.29.1-goqpbgifppele5bt5ewrhcvuuovnx5l6 -> /tmp/adam/spack-stage/spack-stage-0m_w3n1p
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[1860118916110345305:1] [Acquiring]
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[1860118916110345305:1] [Acquired]
==> binutils is already installed in /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-6.3.0/binutils-2.29.1-goqpbgifppele5bt5ewrhcvuuovnx5l6
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[1860118916110345305:1] [Released]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[3520716488889116390:1] [Acquiring]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[3520716488889116390:1] [Acquired]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[3520716488889116390:1] [Released]
==> link /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/openssl-1.0.2n-mg4dd7dwgfg42fzozqcejq7vjwvvj672 -> /tmp/adam/spack-stage/spack-stage-jolcq0zv
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[3520716488889116390:1] [Acquiring]
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[3520716488889116390:1] [Acquired]
==> openssl is already installed in /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-6.3.0/openssl-1.0.2n-mg4dd7dwgfg42fzozqcejq7vjwvvj672
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[3520716488889116390:1] [Released]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[9111565839732578140:1] [Acquiring]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[9111565839732578140:1] [Acquired]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[9111565839732578140:1] [Released]
==> link /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/cmake-3.11.1-7tszdnsrwaxlrknjgohn3txphjgnligx -> /tmp/adam/spack-stage/spack-stage-be_qownv
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[9111565839732578140:1] [Acquiring]
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[9111565839732578140:1] [Acquired]
==> cmake is already installed in /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-6.3.0/cmake-3.11.1-7tszdnsrwaxlrknjgohn3txphjgnligx
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[9111565839732578140:1] [Released]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[745731832071678633:1] [Acquiring]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[745731832071678633:1] [Acquired]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[745731832071678633:1] [Released]
==> link /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/curl-7.59.0-cszl4thivlkvfauym2eqvkijqlrip3ti -> /tmp/adam/spack-stage/spack-stage-efn28mf8
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[745731832071678633:1] [Acquiring]
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[745731832071678633:1] [Acquired]
==> curl is already installed in /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-6.3.0/curl-7.59.0-cszl4thivlkvfauym2eqvkijqlrip3ti
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[745731832071678633:1] [Released]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[4137110530533425660:1] [Acquiring]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[4137110530533425660:1] [Acquired]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[4137110530533425660:1] [Released]
==> link /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/libtool-2.4.6-olj7an3b6ib7qkyi7xl3dcijavzqqckm -> /tmp/adam/spack-stage/spack-stage-ehprpeb4
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[4137110530533425660:1] [Acquiring]
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[4137110530533425660:1] [Acquired]
==> libtool is already installed in /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-6.3.0/libtool-2.4.6-olj7an3b6ib7qkyi7xl3dcijavzqqckm
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[4137110530533425660:1] [Released]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[2861471313054614528:1] [Acquiring]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[2861471313054614528:1] [Acquired]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[2861471313054614528:1] [Released]
==> link /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/util-macros-1.19.1-j5v7t7z3f2eabh63ofv3tt4xl3axgcaa -> /tmp/adam/spack-stage/spack-stage-87yef4hi
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[2861471313054614528:1] [Acquiring]
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[2861471313054614528:1] [Acquired]
==> util-macros is already installed in /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-6.3.0/util-macros-1.19.1-j5v7t7z3f2eabh63ofv3tt4xl3axgcaa
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[2861471313054614528:1] [Released]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[1005697717519611755:1] [Acquiring]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[1005697717519611755:1] [Acquired]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[1005697717519611755:1] [Released]
==> link /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/libpciaccess-0.13.5-dpu6s7oillxnpgxsh2sinsostsgwun3v -> /tmp/adam/spack-stage/spack-stage-oup4f3fi
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[1005697717519611755:1] [Acquiring]
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[1005697717519611755:1] [Acquired]
==> libpciaccess is already installed in /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-6.3.0/libpciaccess-0.13.5-dpu6s7oillxnpgxsh2sinsostsgwun3v
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[1005697717519611755:1] [Released]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[142601053527932287:1] [Acquiring]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[142601053527932287:1] [Acquired]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[142601053527932287:1] [Released]
==> link /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/readline-7.0-ap2t3triih5p5lnqcy4fa543xbm53ibw -> /tmp/adam/spack-stage/spack-stage-kohu0w7x
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[142601053527932287:1] [Acquiring]
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[142601053527932287:1] [Acquired]
==> readline is already installed in /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-6.3.0/readline-7.0-ap2t3triih5p5lnqcy4fa543xbm53ibw
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[142601053527932287:1] [Released]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[6863611065029807916:1] [Acquiring]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[6863611065029807916:1] [Acquired]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[6863611065029807916:1] [Released]
==> link /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/gdbm-1.14.1-x2aohta7cwpfrnfoetcvhx7l2q43bzwe -> /tmp/adam/spack-stage/spack-stage-x2oyft6p
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[6863611065029807916:1] [Acquiring]
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[6863611065029807916:1] [Acquired]
==> gdbm is already installed in /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-6.3.0/gdbm-1.14.1-x2aohta7cwpfrnfoetcvhx7l2q43bzwe
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[6863611065029807916:1] [Released]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[5431469965038050820:1] [Acquiring]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[5431469965038050820:1] [Acquired]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[5431469965038050820:1] [Released]
==> link /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/perl-5.24.1-s3aout7es66atcubwbccordaulsfrmlu -> /tmp/adam/spack-stage/spack-stage-q_dlkam1
==> link /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/resource-cpanm-s3aout7es66atcubwbccordaulsfrmlu -> /tmp/adam/spack-stage/spack-stage-4ubegc2f
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[5431469965038050820:1] [Acquiring]
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[5431469965038050820:1] [Acquired]
==> perl is already installed in /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-6.3.0/perl-5.24.1-s3aout7es66atcubwbccordaulsfrmlu
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[5431469965038050820:1] [Released]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[8236492414840226437:1] [Acquiring]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[8236492414840226437:1] [Acquired]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[8236492414840226437:1] [Released]
==> link /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/autoconf-2.69-4sn4y3glqccqu5qmw5wyau6i3c7zc7kl -> /tmp/adam/spack-stage/spack-stage-eaj5rzw4
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[8236492414840226437:1] [Acquiring]
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[8236492414840226437:1] [Acquired]
==> autoconf is already installed in /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-6.3.0/autoconf-2.69-4sn4y3glqccqu5qmw5wyau6i3c7zc7kl
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[8236492414840226437:1] [Released]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[1296787647320781288:1] [Acquiring]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[1296787647320781288:1] [Acquired]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[1296787647320781288:1] [Released]
==> link /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/automake-1.15.1-ep7dv7jxk3r5byhfwa2nurcxhyncpdjm -> /tmp/adam/spack-stage/spack-stage-1amy3u2a
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[1296787647320781288:1] [Acquiring]
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[1296787647320781288:1] [Acquired]
==> automake is already installed in /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-6.3.0/automake-1.15.1-ep7dv7jxk3r5byhfwa2nurcxhyncpdjm
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[1296787647320781288:1] [Released]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[6559234386500196049:1] [Acquiring]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[6559234386500196049:1] [Acquired]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[6559234386500196049:1] [Released]
==> link /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/numactl-2.0.11-wyhctwblxyw2fedsr2lwr2fiausrtfcc -> /tmp/adam/spack-stage/spack-stage-spyrme2m
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[6559234386500196049:1] [Acquiring]
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[6559234386500196049:1] [Acquired]
==> numactl is already installed in /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-6.3.0/numactl-2.0.11-wyhctwblxyw2fedsr2lwr2fiausrtfcc
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[6559234386500196049:1] [Released]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[4676795304630204240:1] [Acquiring]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[4676795304630204240:1] [Acquired]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[4676795304630204240:1] [Released]
==> link /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/hwloc-1.11.9-qhhkcgqhmvxkdaczhhgghn6vy4l2udlw -> /tmp/adam/spack-stage/spack-stage-4vgx2tr_
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[4676795304630204240:1] [Acquiring]
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[4676795304630204240:1] [Acquired]
==> hwloc is already installed in /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-6.3.0/hwloc-1.11.9-qhhkcgqhmvxkdaczhhgghn6vy4l2udlw
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[4676795304630204240:1] [Released]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[2203894928496600381:1] [Acquiring]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[2203894928496600381:1] [Acquired]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[2203894928496600381:1] [Released]
==> link /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/openmpi-3.0.1-huvz3bl6uzfhvfdqwrmfeykmmboqviff -> /tmp/adam/spack-stage/spack-stage-v45diyjb
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[2203894928496600381:1] [Acquiring]
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[2203894928496600381:1] [Acquired]
==> openmpi is already installed in /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-6.3.0/openmpi-3.0.1-huvz3bl6uzfhvfdqwrmfeykmmboqviff
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[2203894928496600381:1] [Released]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[3165831900327136525:1] [Acquiring]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[3165831900327136525:1] [Acquired]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[3165831900327136525:1] [Released]
==> link /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/sqlite-3.22.0-k7pjnlq3gknbx5pbqqtpri7xkyiu67zc -> /tmp/adam/spack-stage/spack-stage-w_vnncbo
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[3165831900327136525:1] [Acquiring]
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[3165831900327136525:1] [Acquired]
==> sqlite is already installed in /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-6.3.0/sqlite-3.22.0-k7pjnlq3gknbx5pbqqtpri7xkyiu67zc
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[3165831900327136525:1] [Released]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[4708240155827188000:1] [Acquiring]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[4708240155827188000:1] [Acquired]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[4708240155827188000:1] [Released]
==> link /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/python-2.7.14-qkxa547ktzrebyzscnqvezl5mlnomvs4 -> /tmp/adam/spack-stage/spack-stage-7hcac5zr
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[4708240155827188000:1] [Acquiring]
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[4708240155827188000:1] [Acquired]
==> python is already installed in /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-6.3.0/python-2.7.14-qkxa547ktzrebyzscnqvezl5mlnomvs4
==> READ LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[4708240155827188000:1] [Released]
==> Installing julia
==> '/home/Adama-docs/Adam/linux/tmp/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-6.3.0/python-2.7.14-qkxa547ktzrebyzscnqvezl5mlnomvs4/bin/python2.7' '-c' 'from distutils.sysconfig import get_config_var; print get_config_var('"'"'prefix'"'"')'
==> '/home/Adama-docs/Adam/linux/tmp/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-6.3.0/python-2.7.14-qkxa547ktzrebyzscnqvezl5mlnomvs4/bin/python2.7' '-c' 'from distutils.sysconfig import get_config_var; print get_config_var('"'"'prefix'"'"')'
==> Reading config file /home/adam/.spack/linux/mirrors.yaml
==> Deprecation warning: md5 checksums will not be supported in future Spack releases.
==> Using cached archive: /home/Adama-docs/Adam/linux/tmp/spack/var/spack/cache/julia/julia-0.6.2.tar.gz
==> Deprecation warning: md5 checksums will not be supported in future Spack releases.
==> Staging archive: /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/julia-0.6.2-jdn352z55qv6kjotkpxczr2phklrtxu7/julia-0.6.2-full.tar.gz
==> '/bin/tar' '-xf' '/home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/julia-0.6.2-jdn352z55qv6kjotkpxczr2phklrtxu7/julia-0.6.2-full.tar.gz'
==> Created stage in /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/julia-0.6.2-jdn352z55qv6kjotkpxczr2phklrtxu7
==> No patches needed for julia
==> Building julia [Package]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/.lock[7925610364486451665:1] [Acquiring]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/.lock[7925610364486451665:1] [Acquired]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[2624999854097962482:1] [Acquiring]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[2624999854097962482:1] [Acquired]
==> Executing phase: 'install'
==> 'make'
Warning: git information unavailable; versioning information limited
Warning: git information unavailable; versioning information limited
patching file SuiteSparse_config/SuiteSparse_config.h
rm: cannot remove 'scratch/openblas-85636ff1a015d04d3a8f960bc644b85ee5157135/': No such file or directory
patching file driver/others/blas_server.c
patching file cpuid_x86.c
patching file driver/others/dynamic.c
patching file kernel/x86_64/dgemm_kernel_4x8_sandy.S
patching file c_check
patching file kernel/power/casum.c
patching file kernel/power/casum_microk_power8.c
patching file kernel/power/ccopy_microk_power8.c
patching file kernel/power/cswap_microk_power8.c
patching file kernel/power/dasum.c
patching file kernel/power/dasum_microk_power8.c
patching file kernel/power/daxpy.c
patching file kernel/power/daxpy_microk_power8.c
patching file kernel/power/dcopy_microk_power8.c
patching file kernel/power/ddot.c
patching file kernel/power/ddot_microk_power8.c
patching file kernel/power/dgemv_n.c
patching file kernel/power/dgemv_n_microk_power8.c
patching file kernel/power/drot.c
patching file kernel/power/drot_microk_power8.c
patching file kernel/power/dscal.c
patching file kernel/power/dscal_microk_power8.c
patching file kernel/power/dswap_microk_power8.c
patching file kernel/power/sasum.c
patching file kernel/power/sasum_microk_power8.c
patching file kernel/power/scopy_microk_power8.c
patching file kernel/power/sdot.c
patching file kernel/power/sdot_microk_power8.c
patching file kernel/power/srot.c
patching file kernel/power/srot_microk_power8.c
patching file kernel/power/sscal.c
patching file kernel/power/sscal_microk_power8.c
patching file kernel/power/sswap_microk_power8.c
patching file kernel/power/zasum.c
patching file kernel/power/zasum_microk_power8.c
patching file kernel/power/zaxpy.c
patching file kernel/power/zaxpy_microk_power8.c
patching file kernel/power/zcopy_microk_power8.c
patching file kernel/power/zdot.c
Hunk #1 succeeded at 44 (offset 1 line).
patching file kernel/power/zdot_microk_power8.c
patching file kernel/power/zscal.c
patching file kernel/power/zscal_microk_power8.c
patching file kernel/power/zswap_microk_power8.c
make -C scratch/openblas-85636ff1a015d04d3a8f960bc644b85ee5157135/ CC=gcc -m64 FC=gfortran -m64 RANLIB=ranlib TARGET= BINARY=64 USE_THREAD=1 GEMM_MULTITHREADING_THRESHOLD=50 NUM_THREADS=16 NO_AFFINITY=1 DYNAMIC_ARCH=1 INTERFACE64=1 SYMBOLSUFFIX=64_ LIBPREFIX=libopenblas64_ CFLAGS= -O2 FFLAGS= -O2 -fPIC MAKE_NB_JOBS=0
getarch_2nd.c: In function ‘main’:
getarch_2nd.c:12:35: error: ‘SGEMM_DEFAULT_UNROLL_M’ undeclared (first use in this function)
     printf("SGEMM_UNROLL_M=%d\n", SGEMM_DEFAULT_UNROLL_M);
                                   ^~~~~~~~~~~~~~~~~~~~~~
getarch_2nd.c:12:35: note: each undeclared identifier is reported only once for each function it appears in
getarch_2nd.c:13:35: error: ‘SGEMM_DEFAULT_UNROLL_N’ undeclared (first use in this function)
     printf("SGEMM_UNROLL_N=%d\n", SGEMM_DEFAULT_UNROLL_N);
                                   ^~~~~~~~~~~~~~~~~~~~~~
getarch_2nd.c:14:35: error: ‘DGEMM_DEFAULT_UNROLL_M’ undeclared (first use in this function)
     printf("DGEMM_UNROLL_M=%d\n", DGEMM_DEFAULT_UNROLL_M);
                                   ^~~~~~~~~~~~~~~~~~~~~~
getarch_2nd.c:15:35: error: ‘DGEMM_DEFAULT_UNROLL_N’ undeclared (first use in this function)
     printf("DGEMM_UNROLL_N=%d\n", DGEMM_DEFAULT_UNROLL_N);
                                   ^~~~~~~~~~~~~~~~~~~~~~
getarch_2nd.c:19:35: error: ‘CGEMM_DEFAULT_UNROLL_M’ undeclared (first use in this function)
     printf("CGEMM_UNROLL_M=%d\n", CGEMM_DEFAULT_UNROLL_M);
                                   ^~~~~~~~~~~~~~~~~~~~~~
getarch_2nd.c:20:35: error: ‘CGEMM_DEFAULT_UNROLL_N’ undeclared (first use in this function)
     printf("CGEMM_UNROLL_N=%d\n", CGEMM_DEFAULT_UNROLL_N);
                                   ^~~~~~~~~~~~~~~~~~~~~~
getarch_2nd.c:21:35: error: ‘ZGEMM_DEFAULT_UNROLL_M’ undeclared (first use in this function)
     printf("ZGEMM_UNROLL_M=%d\n", ZGEMM_DEFAULT_UNROLL_M);
                                   ^~~~~~~~~~~~~~~~~~~~~~
getarch_2nd.c:22:35: error: ‘ZGEMM_DEFAULT_UNROLL_N’ undeclared (first use in this function)
     printf("ZGEMM_UNROLL_N=%d\n", ZGEMM_DEFAULT_UNROLL_N);
                                   ^~~~~~~~~~~~~~~~~~~~~~
getarch_2nd.c:69:50: error: ‘SGEMM_DEFAULT_Q’ undeclared (first use in this function)
     printf("#define SLOCAL_BUFFER_SIZE\t%ld\n", (SGEMM_DEFAULT_Q * SGEMM_DEFAULT_UNROLL_N * 4 * 1 *  sizeof(float)));
                                                  ^~~~~~~~~~~~~~~
getarch_2nd.c:70:50: error: ‘DGEMM_DEFAULT_Q’ undeclared (first use in this function)
     printf("#define DLOCAL_BUFFER_SIZE\t%ld\n", (DGEMM_DEFAULT_Q * DGEMM_DEFAULT_UNROLL_N * 2 * 1 *  sizeof(double)));
                                                  ^~~~~~~~~~~~~~~
getarch_2nd.c:71:50: error: ‘CGEMM_DEFAULT_Q’ undeclared (first use in this function)
     printf("#define CLOCAL_BUFFER_SIZE\t%ld\n", (CGEMM_DEFAULT_Q * CGEMM_DEFAULT_UNROLL_N * 4 * 2 *  sizeof(float)));
                                                  ^~~~~~~~~~~~~~~
getarch_2nd.c:72:50: error: ‘ZGEMM_DEFAULT_Q’ undeclared (first use in this function)
     printf("#define ZLOCAL_BUFFER_SIZE\t%ld\n", (ZGEMM_DEFAULT_Q * ZGEMM_DEFAULT_UNROLL_N * 2 * 2 *  sizeof(double)));
                                                  ^~~~~~~~~~~~~~~
make[2]: *** [getarch_2nd] Error 1
Makefile:123: *** OpenBLAS: Detecting CPU failed. Please set TARGET explicitly, e.g. make TARGET=your_cpu_target. Please read README for the detail..  Stop.
*** Clean the OpenBLAS build with 'make -C deps clean-openblas'. Rebuild with 'make OPENBLAS_USE_THREAD=0' if OpenBLAS had trouble linking libpthread.so, and with 'make OPENBLAS_TARGET_ARCH=NEHALEM' if there were errors building SandyBridge support. Both these options can also be used simultaneously. ***
/tmp/adam/spack-stage/spack-stage-1q4cgpcb/julia-0.6.2/deps/blas.mk:111: recipe for target 'scratch/openblas-85636ff1a015d04d3a8f960bc644b85ee5157135/build-compiled' failed
make[1]: *** [scratch/openblas-85636ff1a015d04d3a8f960bc644b85ee5157135/build-compiled] Error 1
Makefile:85: recipe for target 'julia-deps' failed
make: *** [julia-deps] Error 2
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/opt/spack/.spack-db/prefix_lock[2624999854097962482:1] [Released]
==> WRITE LOCK: /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/.lock[7925610364486451665:1] [Released]
==> Error: ProcessError: Command exited with status 2:
    'make'

17 errors found in build log:
  >> 6     rm: cannot remove 'scratch/openblas-85636ff1a015d04d3a8f960bc644b85ee5157135/': No such file or directory
     7     patching file driver/others/blas_server.c
     8     patching file cpuid_x86.c
     9     patching file driver/others/dynamic.c
     10    patching file kernel/x86_64/dgemm_kernel_4x8_sandy.S
     11    patching file c_check
     12    patching file kernel/power/casum.c

     ...

     47    patching file kernel/power/zdot_microk_power8.c
     48    patching file kernel/power/zscal.c
     49    patching file kernel/power/zscal_microk_power8.c
     50    patching file kernel/power/zswap_microk_power8.c
     51    make -C scratch/openblas-85636ff1a015d04d3a8f960bc644b85ee5157135/ CC=gcc -m64 FC=gfortran -m64 RANLIB=ranlib TARGET= BINARY=64 USE_THREAD=1 GEMM_MULTITHREADING_THRESHOLD=50 NUM_THREADS=16 NO_AFFINITY=1 DYNAMIC_ARCH=1 INTERFACE64=1 SYMBOLSUFFIX=64_ LIBPREFIX=l
           ibopenblas64_ CFLAGS= -O2 FFLAGS= -O2 -fPIC MAKE_NB_JOBS=0
     52    getarch_2nd.c: In function ‘main’:
  >> 53    getarch_2nd.c:12:35: error: ‘SGEMM_DEFAULT_UNROLL_M’ undeclared (first use in this function)
     54         printf("SGEMM_UNROLL_M=%d\n", SGEMM_DEFAULT_UNROLL_M);
     55                                       ^~~~~~~~~~~~~~~~~~~~~~
     56    getarch_2nd.c:12:35: note: each undeclared identifier is reported only once for each function it appears in
  >> 57    getarch_2nd.c:13:35: error: ‘SGEMM_DEFAULT_UNROLL_N’ undeclared (first use in this function)
     58         printf("SGEMM_UNROLL_N=%d\n", SGEMM_DEFAULT_UNROLL_N);
     59                                       ^~~~~~~~~~~~~~~~~~~~~~
  >> 60    getarch_2nd.c:14:35: error: ‘DGEMM_DEFAULT_UNROLL_M’ undeclared (first use in this function)
     61         printf("DGEMM_UNROLL_M=%d\n", DGEMM_DEFAULT_UNROLL_M);
     62                                       ^~~~~~~~~~~~~~~~~~~~~~
  >> 63    getarch_2nd.c:15:35: error: ‘DGEMM_DEFAULT_UNROLL_N’ undeclared (first use in this function)
     64         printf("DGEMM_UNROLL_N=%d\n", DGEMM_DEFAULT_UNROLL_N);
     65                                       ^~~~~~~~~~~~~~~~~~~~~~
  >> 66    getarch_2nd.c:19:35: error: ‘CGEMM_DEFAULT_UNROLL_M’ undeclared (first use in this function)
     67         printf("CGEMM_UNROLL_M=%d\n", CGEMM_DEFAULT_UNROLL_M);
     68                                       ^~~~~~~~~~~~~~~~~~~~~~
  >> 69    getarch_2nd.c:20:35: error: ‘CGEMM_DEFAULT_UNROLL_N’ undeclared (first use in this function)
     70         printf("CGEMM_UNROLL_N=%d\n", CGEMM_DEFAULT_UNROLL_N);
     71                                       ^~~~~~~~~~~~~~~~~~~~~~
  >> 72    getarch_2nd.c:21:35: error: ‘ZGEMM_DEFAULT_UNROLL_M’ undeclared (first use in this function)
     73         printf("ZGEMM_UNROLL_M=%d\n", ZGEMM_DEFAULT_UNROLL_M);
     74                                       ^~~~~~~~~~~~~~~~~~~~~~
  >> 75    getarch_2nd.c:22:35: error: ‘ZGEMM_DEFAULT_UNROLL_N’ undeclared (first use in this function)
     76         printf("ZGEMM_UNROLL_N=%d\n", ZGEMM_DEFAULT_UNROLL_N);
     77                                       ^~~~~~~~~~~~~~~~~~~~~~
  >> 78    getarch_2nd.c:69:50: error: ‘SGEMM_DEFAULT_Q’ undeclared (first use in this function)
     79         printf("#define SLOCAL_BUFFER_SIZE\t%ld\n", (SGEMM_DEFAULT_Q * SGEMM_DEFAULT_UNROLL_N * 4 * 1 *  sizeof(float)));
     80                                                      ^~~~~~~~~~~~~~~
  >> 81    getarch_2nd.c:70:50: error: ‘DGEMM_DEFAULT_Q’ undeclared (first use in this function)
     82         printf("#define DLOCAL_BUFFER_SIZE\t%ld\n", (DGEMM_DEFAULT_Q * DGEMM_DEFAULT_UNROLL_N * 2 * 1 *  sizeof(double)));
     83                                                      ^~~~~~~~~~~~~~~
  >> 84    getarch_2nd.c:71:50: error: ‘CGEMM_DEFAULT_Q’ undeclared (first use in this function)
     85         printf("#define CLOCAL_BUFFER_SIZE\t%ld\n", (CGEMM_DEFAULT_Q * CGEMM_DEFAULT_UNROLL_N * 4 * 2 *  sizeof(float)));
     86                                                      ^~~~~~~~~~~~~~~
  >> 87    getarch_2nd.c:72:50: error: ‘ZGEMM_DEFAULT_Q’ undeclared (first use in this function)
     88         printf("#define ZLOCAL_BUFFER_SIZE\t%ld\n", (ZGEMM_DEFAULT_Q * ZGEMM_DEFAULT_UNROLL_N * 2 * 2 *  sizeof(double)));
     89                                                      ^~~~~~~~~~~~~~~
  >> 90    make[2]: *** [getarch_2nd] Error 1
     91    Makefile:123: *** OpenBLAS: Detecting CPU failed. Please set TARGET explicitly, e.g. make TARGET=your_cpu_target. Please read README for the detail..  Stop.
     92    *** Clean the OpenBLAS build with 'make -C deps clean-openblas'. Rebuild with 'make OPENBLAS_USE_THREAD=0' if OpenBLAS had trouble linking libpthread.so, and with 'make OPENBLAS_TARGET_ARCH=NEHALEM' if there were errors building SandyBridge support. Both these
            options can also be used simultaneously. ***
  >> 93    /tmp/adam/spack-stage/spack-stage-1q4cgpcb/julia-0.6.2/deps/blas.mk:111: recipe for target 'scratch/openblas-85636ff1a015d04d3a8f960bc644b85ee5157135/build-compiled' failed
  >> 94    make[1]: *** [scratch/openblas-85636ff1a015d04d3a8f960bc644b85ee5157135/build-compiled] Error 1
     95    Makefile:85: recipe for target 'julia-deps' failed
  >> 96    make: *** [julia-deps] Error 2

See build log for details:
  /home/Adama-docs/Adam/linux/tmp/spack/var/spack/stage/julia-0.6.2-jdn352z55qv6kjotkpxczr2phklrtxu7/julia-0.6.2/spack-build.out
Traceback (most recent call last):
  File "/home/Adama-docs/Adam/linux/tmp/spack/lib/spack/spack/build_environment.py", line 666, in child_process
    return_value = function()
  File "/home/Adama-docs/Adam/linux/tmp/spack/lib/spack/spack/package.py", line 1527, in build_process
    phase(self.spec, self.prefix)
  File "/home/Adama-docs/Adam/linux/tmp/spack/lib/spack/spack/package.py", line 119, in phase_wrapper
    phase(spec, prefix)
  File "/home/Adama-docs/Adam/linux/tmp/spack/var/spack/repos/builtin/packages/julia/package.py", line 151, in install
    make()
  File "/home/Adama-docs/Adam/linux/tmp/spack/lib/spack/spack/build_environment.py", line 126, in __call__
    return super(MakeExecutable, self).__call__(*args, **kwargs)
  File "/home/Adama-docs/Adam/linux/tmp/spack/lib/spack/spack/util/executable.py", line 183, in __call__
    proc.returncode, cmd_line)
spack.util.executable.ProcessError: Command exited with status 2:
    'make'
```
### Information on your system
Ubuntu 16.04 64bit with current patches and updates. No custom spack packages nor modules. OpenBLAS installs OK with spack, if invoked by `spack install openblas`

**Comment by adamjstewart** [comment]
@eschnett?

**Comment by jamesETsmith** [comment]
I see this problem when building Julia 1.6.2 using a `spack` installed version of OpenBLAS (0.3.17) on MacOS (11.6). Was there ever a fix for this? I know #6941 is already a similar issue to this one and I didn't want to open another overlapping issue (but if that's preferred just let me know).

**Comment by alalazo** [comment]
Closing, since there has been no activity in a while. Feel free to open a new issue if the problem persists.
